core/arch/arm/instr_create.h

```diff
@@ -115,6 +115,17 @@
 #define XINST_CREATE_load(dc, r, m)  INSTR_CREATE_ldr((dc), (r), (m))
 
 /**
+ * This platform-independent macro creates an instr_t which takes a 4-byte
+ * destination and performs a zero-extend on x86, or an "effective" 1-byte
+ * load. This performs a traditional 1-byte load on ARM, which is consistent
+ * with the behaviour on x86.
```

Comment:
OK, though I'm not sure we need to talk about x86 or ARM: we want to present this as a platform-independent method to load 1 byte from memory, zero-extend it to 4 bytes, and write it to a register.  That's all the tool writer needs to know: underlying details like this being a regular ARM load but a special x86 load seem superfluous and may just confuse the user.